### ISSUE TITLE
5.0 volfraction in S(Q)

### ISSUE THREAD

**Issue by RichardHeenan** [SasView/sasmodels#232]
I am opening a new ticket for this, though it has in part been covered by a number of other tickets related to beta approx project and P(Q)S(Q) (most of which can now be closed).

The radius_effective and beta approx parts now work, though as usual a lot more unit tests are needed.   [Also some issues with vesicle mode.]

The final part of the "beta approx, plus improve S(Q)" project was not started as the UK student, Torin, ran out of time. It was intended that we sort out the the use of volfraction and scale when using S(Q).

What we have currently in v5.0 regarding scale & volfraction is the same as v4.2 , without S(Q) then:
I(Q) = scale.P(radius,Q)    where scale is volume fraction if working in absolute units
with S(Q) then:
I(Q) = scale.Svolfrac.P(radius,Q).S(Q,radius_effective,volfraction_effective) 

This is confusing for users as the value and meaning of scale changes on adding an S(Q), it would be much better is this did not happen.

We should in v5.0 add a third drop down in the fitting gui to choose whether to constrain volume fraction in S(Q) or let it freely adjust, i.e. volfraction_effective_mode=1 for constrained, 0 for freely adusting.  The drop down could appear alongside the one for radius_effective_mode. 

It has been suggested that any P(Q) which can be multiplied by an S(Q) should have an explicit volfraction parameter added, but I don't think that is actually necessary.

I would **propose** -     I(Q)=scale.P(radius,Q).S(Q,radius_effective,volfraction_effective)

Where if working in absolute units, then just as in v4.2,  scale = phi the volume fraction of particles, but now Svolfrac is only affects the shape of S(Q) and not the overall height of I(Q).

In the main fitting gui selecting volfraction_effective_mode=1, constrains Svolfrac = scale
                                                   volfraction_effective_mode=0,  allows Svolfrac to be fitted

In sum/multi plugin models then volfraction_effective_mode defaults to 0, exactly as v4.2

If loading a P(Q)S(Q) from sasview 4, then scale has to be set to scale_v4*volfraction

I think that this should be done before releasing 5.1 to the general public to avoid confusion later.

Would this work ? - your opinions please.

We could **instead** add volfraction explicitly to all P(Q) models that can take an S(Q), so that 

I(Q)=scale.volfraction.P(R,Q,volfraction).S(Q,Reff,Svolfrac)

But I  don't think that there any P(Q) that actually need volume fraction as a genuine parameter ? 

NOTES
 (i) volfraction_mode = 1 requires special constraints as v4.2 for vesicle and hollow_cylinder models which are scaled by the shell rather than exterior volume 
(ii) Some changes would be required to the results checks of the new @S unit tests, and 
(iii) likely also to sasfit_compare_new.py

**Comment by pkienzle** [comment]
[sasview #2794](https://github.com/SasView/sasview/issues/2794) shows that confusion persists.

Including Ryan's example of core-shell sphere vs. vesicle model in the docs would help.

We could also modify the description of volfraction for vesicle to say that it is uses the volume fraction of the enclosed volume when computing the structure factor. (See also [sasview #3973](https://github.com/SasView/sasview/issues/3973))
